packages/core/src/services/oidcServices.js

```diff
@@ -1,20 +1,39 @@
 import { oidcLog } from './loggerService';
 
 let userRequested = false;
-export const isRequireAuthentication = (oidcUser, isForce) =>
-  isForce || !oidcUser || oidcUser.expired;
 
-export const authenticateUser = (userManager, location) => async (isForce = false) => {
-  if (!userManager || !userManager.getUser) {
+export const isRequireAuthentication = (user, isForce) =>
+  isForce || !user || (user && user.expired === true);
+
+export const isRequireSignin = (oidcUser, isForce) => isForce || !oidcUser;
+
+export const authenticateUser = (userManager, location, history, user = null) => async (
+  isForce = false
+) => {
+  let oidcUser = user;
+  if (!oidcUser) {
+    oidcUser = await userManager.getUser();
+  }
+  if (userRequested) {
     return;
   }
-  const oidcUser = await userManager.getUser();
-  if (isRequireAuthentication(oidcUser, isForce) && !userRequested) {
+  const url = location.pathname + (location.search || '');
+
+  if (isRequireSignin(oidcUser, isForce)) {
     oidcLog.info('authenticate user...');
-    const url = location.pathname + (location.search || '');
     userRequested = true;
     await userManager.signinRedirect({ data: { url } });
     userRequested = false;
+  } else if (oidcUser && oidcUser.expired) {
```

Comment:
We never had to do that, don't you think that the problem is deeper ?

Reply:
Je vais essayer de finir cette pr la semaine prochaine. J'avais presque fini.

Reply:
Oui c'est ce qui uniformise le comportement des 2 librairies.